### PR TITLE
chore: dependency updates

### PR DESCRIPTION
**Pull request by TimBeyer**
**What this PR does / why we need it**:
Bumps a whole lot of dependencies now that we're supporting ESM modules

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/server/server.ts [hunk 1]
```diff
@@ -224,8 +222,7 @@ export class GardenServer extends EventEmitter {
         try {
           this.port = await getPort({
             host: hostname,
-            port: defaultServerPort,
-            portRange: [defaultServerPort + 1, defaultServerPort + 99],
+            port: portNumbers(defaultServerPort, defaultServerPort + 99),
```
Comment (stefreak) [inline core/src/server/server.ts:225]: @vvagaytsev does this change look good to you?

File: core/src/util/artifacts.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ export async function copyArtifacts({
   const { default: cpy } = await import("cpy")
 
   try {
-    files = await cpy("**/*", garden.artifactsPath, { cwd: artifactsPath, parents: true })
+    files = await cpy("./**/*", garden.artifactsPath, { cwd: artifactsPath })
```
Comment (stefreak) [inline core/src/util/artifacts.ts:80]: What did `parents: true` do?

**Review by stefreak — COMMENTED** [review]
Looks good to me so far; There is still one test failure and I commented on the changes around `cpy` and `getPort`

File: core/src/util/artifacts.ts (hunk 2 above)
Comment (TimBeyer) [reply core/src/util/artifacts.ts:80]: It's basically the default behavior now. It ensured that it preserved the directory tree while recursing.

**Review by stefreak — APPROVED** [review]
Awesome! Now we're up-to-date.

File: core/src/server/server.ts (hunk 1 above)
Comment (vvagaytsev) [reply core/src/server/server.ts:225]: Yes, that looks totally good! The usage of `get-port-please` was a temporary solution.

**Comment by vvagaytsev** [comment]
This also supersedes the temporary fixes implemented in #4984 and #5163. Great job, thank you so much! 👍 💯 🚀

**Pull request merged**